tests/phpunit/tests/theme/wpThemeJson.php

```diff
@@ -4434,4 +4434,75 @@ public function test_get_stylesheet_returns_outline_styles() {
 		$expected = $base_styles . $element_styles;
 		$this->assertSame( $expected, $theme_json->get_stylesheet() );
 	}
+
+	/**
+	 * @ticket 57559
+	 */
+	public function test_shadow_preset_styles() {
+		$theme_json = new WP_Theme_JSON(
+			array(
+				'version'  => WP_Theme_JSON::LATEST_SCHEMA,
+				'settings' => array(
+					'shadow' => array(
+						'presets' => array(
+							array(
+								'slug'   => 'natural',
+								'shadow' => '5px 5px 5px 0 black',
+							),
+							array(
+								'slug'   => 'sharp',
+								'shadow' => '5px 5px black',
+							),
+						),
+					),
+				),
+			)
+		);
+
+		$styles = 'body{--wp--preset--shadow--natural: 5px 5px 5px 0 black;--wp--preset--shadow--sharp: 5px 5px black;}';
+		$this->assertEquals( $styles, $theme_json->get_stylesheet(), 'Returned of "::get_stylesheet" does not match expectations' );
+		$this->assertEquals( $styles, $theme_json->get_stylesheet( array( 'variables' ) ), 'Returned of "::get_stylesheet" does not match expectations' );
```

Comment:
* use `assertSame()`
* Failure messages need to be different to identify which assertion failed.

Commit coming.

Reply:
Resolved in https://github.com/WordPress/wordpress-develop/pull/3915/commits/5e3e53642a639a3a6d07e8d0f9dbc086b96c68c9.